controllers/openstackdeploy_controller.go

```diff
@@ -0,0 +1,241 @@
+
+		op, err := controllerutil.CreateOrUpdate(context.TODO(), r.Client, job, func() error {
+			err := controllerutil.SetControllerReference(instance, job, r.Scheme)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		})
+		if err != nil && !k8s_errors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
+		if op != controllerutil.OperationResultNone {
+			cond.Message = fmt.Sprintf("Job successfully created/updated - operation: %s", string(op))
+			cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.DeployCondReasonJobCreated)
+			cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.DeployCondTypeInitializing)
+			common.LogForObject(r, cond.Message, instance)
+
+			return ctrl.Result{RequeueAfter: time.Second * 5}, nil
+		}
+
+		// configVersion Hash changed while job was running (NOTE: configVersion is only ENV in the job)
+		common.LogForObject(r, fmt.Sprintf("Job Hash : %s", job.Spec.Template.Spec.Containers[0].Env[0].Value), instance)
+		if instance.Spec.ConfigVersion != job.Spec.Template.Spec.Containers[0].Env[0].Value {
+			_, err = common.DeleteJob(job, r.Kclient, r.Log)
+			if err != nil {
+				cond.Message = err.Error()
+				cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.DeployCondReasonJobDelete)
+				cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.DeployCondTypeError)
+				err = common.WrapErrorForObject(cond.Message, instance, err)
+
+				return ctrl.Result{}, err
+			}
+
+			cond.Message = "ConfigVersion has changed. Requeing to start again..."
+			cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.DeployCondReasonCMUpdated)
+			cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.DeployCondTypeInitializing)
+			common.LogForObject(r, cond.Message, instance)
+
+			return ctrl.Result{RequeueAfter: time.Second * 5}, nil
+
+		}
+
+		requeue, err := common.WaitOnJob(job, r.Client, r.Log)
+		common.LogForObject(r, "Deploying OpenStack...", instance)
+
+		if err != nil {
+			// the job failed in error
+			cond.Message = "Job failed... Please check job/pod logs."
+			cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.DeployCondReasonJobFailed)
+			cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.DeployCondTypeError)
+			err = common.WrapErrorForObject(cond.Message, instance, err)
+
+			return ctrl.Result{}, err
+		} else if requeue {
+			cond.Message = "Waiting on OpenStack Deployment..."
+			cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.DeployCondReasonConfigCreate)
+			cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.DeployCondTypeRunning)
+			common.LogForObject(r, cond.Message, instance)
+
+			return ctrl.Result{RequeueAfter: time.Second * 10}, nil
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// DeployGetLabels -
+func DeployGetLabels(name string) map[string]string {
+	return map[string]string{"owner": "osp-director-operator", "cr": name, "app": "deploy"}
+}
```

Comment:
looks like this is not being used?

Reply:
thanks, removed in latest set of commits